dashboard/app/models/script_level.rb

```diff
@@ -133,9 +133,30 @@ def summarize
     summary
   end
 
+  SCRIPT_LEVEL_CACHE_KEY = 'script-level-cache'
+
+  def self.script_level_cache_to_cache
+    Rails.cache.write(SCRIPT_LEVEL_CACHE_KEY, script_level_cache_from_db)
+  end
+
+  def self.script_level_cache_from_cache
+    ScriptLevel.connection
+    [ScriptLevel, Level, Game, Concept, Callout, Video,
+     Artist, Blockly].each(&:new) # make sure all possible loaded objects are completely loaded
+    Rails.cache.read SCRIPT_LEVEL_CACHE_KEY
+  end
+
+  def self.script_level_cache_from_db
+    ScriptLevel.includes([{level: [:game, :concepts]}, :script]).index_by(&:id)
+  end
+
+  def self.script_level_map
+    @@script_level_map ||=
+        script_level_cache_from_cache || script_level_cache_from_db
+  end
+
   def self.cache_find(id)
-    @@script_level_map ||= ScriptLevel.includes([{level: [:game, :concepts]}, :script]).index_by(&:id)
-    @@script_level_map[id]
+    script_level_map[id]
```

Comment:
maybe this should act like the Script cache and be disabled for levelbuilder environments. Probably hasn't caused a ton of problems because milestone via the script_level_id is not used very much on levelbuilders (since its easier to get to the level with milestone via the level when you are editing levels) -- but I don't see any benefit to using the cache there.